test/mpi/main.cc

```diff
@@ -376,13 +383,18 @@ void ProcessArgs(int argc, char** argv)
         case 'i':
             iterations = std::stoi(optarg);
             break;
+#ifdef HAVE_CUDA
+        case 'S':
+            test_cuda_set_device = (test_set_cuda_device_t)std::stoi(optarg);
+            break;
+#endif
```

Comment:
minor, add #else for HAVE_CUDA and print something like "test wasn't compiled with cuda support"

Reply:
This option will be disabled If test compiled without CUDA, so the user cannot specify this. Do you want the option should be visible anyway?

Reply:
ok